File: android-app/android-app/app/src/main/java/com/pocketree/app/Badge.kt
package com.pocketree.app

import com.google.gson.annotations.SerializedName

data class Badge (
    @SerializedName("badgeID") val badgeID: Int,
    @SerializedName("badgeName") val badgeName: String,
    @SerializedName("badgeDescription") val badgeDescription: String,
    @SerializedName("badgeImageUrl") val badgeImageUrl: String,
    val criteriaType: String,
    val requiredDifficulty: String,
    val requiredCount: Int
)

File: api/Pocketree.Api/Models/Entities/Badge.cs
using System.ComponentModel.DataAnnotations;

namespace ADproject.Models.Entities
{
    public class Badge
    {
        [Key]
        public int BadgeID { get; set; }
        [Required, StringLength(50)]
        public string BadgeName { get; set; }
        [Required, StringLength(255)]
        public string Description { get; set; }
        public string BadgeImageURL { get; set; }
        [Required, StringLength(20)]
        public string CriteriaType { get; set; }
        [Required, StringLength(10)]
        public string RequiredDifficulty { get; set; }
        [Required]
        public int RequiredCount { get; set; }
    
        // Navigation Property
        public virtual ICollection<UserBadge> UserBadges { get; set; }
    }
}
